diff -Naur sysrepo-devel.org/CMakeLists.txt sysrepo-devel/CMakeLists.txt
--- sysrepo-devel.org/CMakeLists.txt	2020-04-16 00:54:51.000000000 +1000
+++ sysrepo-devel/CMakeLists.txt	2020-04-16 11:13:47.769054003 +1000
@@ -204,9 +204,8 @@
 
 # dependencies
 # librt (shm_open, shm_unlink, not required on OSX or QNX)
-find_library(LIBRT rt)
-if(LIBRT)
-    target_link_libraries(sysrepo ${LIBRT})
+if(NOT APPLE)
+    target_link_libraries(sysrepo rt)
 endif()
 
 # atomic
