diff -Naur Netopeer2-devel-server.org/server/main.c Netopeer2-devel-server/server/main.c
--- Netopeer2-devel-server.org/server/main.c	2020-03-18 19:17:28.000000000 +1100
+++ Netopeer2-devel-server/server/main.c	2020-03-19 17:53:27.454231238 +1100
@@ -909,7 +909,7 @@
         }
 
         /* listen for incoming requests on active NETCONF sessions */
-        rc = nc_ps_poll(np2srv.nc_ps, 10, &ncs);
+        rc = nc_ps_poll(np2srv.nc_ps, 100, &ncs);
 
         if ((rc & (NC_PSPOLL_NOSESSIONS | NC_PSPOLL_TIMEOUT | NC_PSPOLL_ERROR)) && !(rc & NC_PSPOLL_SESSION_TERM)) {
             /* if there is no active session, timeout, or an error, rest for a while */
