diff -Naur Netopeer2-0.7-r2.org/keystored/CMakeLists.txt Netopeer2-0.7-r2/keystored/CMakeLists.txt
--- Netopeer2-0.7-r2.org/keystored/CMakeLists.txt	2019-06-27 16:43:10.000000000 +1000
+++ Netopeer2-0.7-r2/keystored/CMakeLists.txt	2019-07-11 16:15:08.946822950 +1000
@@ -115,7 +115,7 @@
 option(MODEL_INSTALL "Enable yang model installation" ON)
 if (MODEL_INSTALL)
     install(CODE "
-        set(ENV{SYSREPOCTL} ${SYSREPOCTL_EXECUTABLE})
+        set(ENV{SYSREPOCTL} \"${SYSREPOCTL_EXECUTABLE}\")
         set(ENV{SYSREPOCTL_ROOT_PERMS} SYSREPOCTL_ROOT_PERMS)
         execute_process(COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/model-install.sh)")
 endif()
@@ -138,7 +138,7 @@
         message(FATAL_ERROR "Unable to find ssh-keygen, set SSH_KEYGEN_EXECUTABLE manually.")
     endif()
     install(CODE "
-        set(ENV{SYSREPOCFG} ${SYSREPOCFG_EXECUTABLE})
+        set(ENV{SYSREPOCFG} \"${SYSREPOCFG_EXECUTABLE}\")
         set(ENV{CHMOD} ${CHMOD_EXECUTABLE})
         set(ENV{OPENSSL} ${OPENSSL_EXECUTABLE})
         set(ENV{SSH_KEYGEN} ${SSH_KEYGEN_EXECUTABLE})
diff -Naur Netopeer2-0.7-r2.org/keystored/scripts/model-install.sh Netopeer2-0.7-r2/keystored/scripts/model-install.sh
--- Netopeer2-0.7-r2.org/keystored/scripts/model-install.sh	2019-06-27 16:43:10.000000000 +1000
+++ Netopeer2-0.7-r2/keystored/scripts/model-install.sh	2019-07-11 16:15:57.882860267 +1000
@@ -5,26 +5,15 @@
 shopt -s failglob
 
 local_path=$(dirname $0)
-
 : ${SYSREPOCTL:=sysrepoctl}
 : ${SYSREPOCTL_ROOT_PERMS:=-o root:root -p 600}
 : ${YANG_DIR:=$local_path/../../modules}
 
-is_yang_module_installed() {
-    module=$1
-
-    $SYSREPOCTL -l | grep --count "^$module [^|]*|[^|]*| Installed .*$" > /dev/null
-}
-
 install_yang_module() {
     module=$1
 
-    if ! is_yang_module_installed $module; then
-        echo "- Installing module $module..."
-        $SYSREPOCTL -i -g ${YANG_DIR}/$module.yang $SYSREPOCTL_ROOT_PERMS
-    else
-        echo "- Module $module already installed."
-    fi
+    echo "- Installing module $module..."
+    $SYSREPOCTL -i -g ${YANG_DIR}/$module.yang $SYSREPOCTL_ROOT_PERMS
 }
 
 install_yang_module ietf-x509-cert-to-name
diff -Naur Netopeer2-0.7-r2.org/server/CMakeLists.txt Netopeer2-0.7-r2/server/CMakeLists.txt
--- Netopeer2-0.7-r2.org/server/CMakeLists.txt	2019-06-27 16:43:10.000000000 +1000
+++ Netopeer2-0.7-r2/server/CMakeLists.txt	2019-07-11 15:34:24.290554617 +1000
@@ -202,8 +202,8 @@
     endif()
 
     install(CODE "
-        set(ENV{SYSREPOCTL} ${SYSREPOCTL_EXECUTABLE})
-        set(ENV{SYSREPOCFG} ${SYSREPOCFG_EXECUTABLE})
+        set(ENV{SYSREPOCTL} \"${SYSREPOCTL_EXECUTABLE}\")
+        set(ENV{SYSREPOCFG} \"${SYSREPOCFG_EXECUTABLE}\")
         set(ENV{SYSREPOCTL_ROOT_PERMS} SYSREPOCTL_ROOT_PERMS)
         execute_process(COMMAND ${PROJECT_BINARY_DIR}/model-install.sh)")
 endif()
diff -Naur Netopeer2-0.7-r2.org/server/scripts/model-install.sh.in Netopeer2-0.7-r2/server/scripts/model-install.sh.in
--- Netopeer2-0.7-r2.org/server/scripts/model-install.sh.in	2019-06-27 16:43:10.000000000 +1000
+++ Netopeer2-0.7-r2/server/scripts/model-install.sh.in	2019-07-11 16:14:13.370780970 +1000
@@ -3,7 +3,6 @@
 set -eu -o pipefail
 
 shopt -s failglob
-
 : ${SYSREPOCTL:=sysrepoctl}
 : ${SYSREPOCFG:=sysrepocfg}
 : ${SYSREPOCTL_ROOT_PERMS:=-o root:root -p 600}
@@ -19,24 +18,16 @@
 install_yang_module() {
     module=$1
 
-    if ! is_yang_module_installed $module; then
-        echo "- Installing module $module..."
-        $SYSREPOCTL -i -g ${YANG_DIR}/$module.yang $SYSREPOCTL_ROOT_PERMS
-    else
-        echo "- Module $module already installed."
-    fi
+    echo "- Installing module $module..."
+    $SYSREPOCTL -i -g ${YANG_DIR}/$module.yang $SYSREPOCTL_ROOT_PERMS
 }
 
 enable_yang_module_feature() {
     module=$1
     feature=$2
 
-    if ! $SYSREPOCTL -l | grep --count "^$module [^|]*|[^|]*|[^|]*|[^|]*|[^|]*|[^|]*|.* $feature.*$" > /dev/null; then
-        echo "- Enabling feature $feature in $module..."
-        $SYSREPOCTL -m $module -e $feature
-    else
-        echo "- Feature $feature in $module already enabled."
-    fi
+    echo "- Enabling feature $feature in $module..."
+    $SYSREPOCTL -m $module -e $feature
 }
 
 install_yang_module ietf-netconf
